.pipelines/.vsts-vhd-windows-automation.yaml

```diff
@@ -7,24 +8,74 @@ parameters:
 - name: ImageBump
   displayName: Image Bump + Branch Cutting
   type: boolean
-  default: true
+  default: false
 - name: GithubUserName
   displayName: The user name of the owner to push changes
   type: string
+- name: ArtifactTrigger
+  displayName: SIG Artifact Trigger
+  type: boolean
+  default: false
+
+stages:
+  - stage: bump_version
+    dependsOn: []
+    condition: eq('${{ parameters.ImageBump }}', true)
+    jobs:
+    - job: bump
+      timeoutInMinutes: 180
+      steps:
+        - bash: |
+                az extension add -n azure-devops
+                echo $MAPPED_ADO_PAT | az devops login --organization=https://dev.azure.com/msazure
+                az devops configure --defaults organization=https://dev.azure.com/msazure project=CloudNativeCompute
+          env:
+                MAPPED_ADO_PAT: $(ADO_PAT)
+          displayName: 'az devops login'
+        - bash: |
+                echo "Bumping windows VHD base image version"
+                /bin/bash vhdbuilder/scripts/windows/automate_version_bump.sh $MAPPED_GITHUB_PAT '${{ parameters.GithubUserName }}' "$CHERRY_PICK_COMMIT_ID"
+          env:
+                MAPPED_GITHUB_PAT: $(GITHUB_PAT)
+          displayName: 'Image Version Bumping'
+          condition: eq('${{ parameters.ImageBump }}', true)
 
-steps:
-- bash: |
-        az extension add -n azure-devops
-        echo $MAPPED_ADO_PAT | az devops login --organization=https://dev.azure.com/msazure
-        az devops configure --defaults organization=https://dev.azure.com/msazure project=CloudNativeCompute
-  env:
-        MAPPED_ADO_PAT: $(ADO_PAT)
-  displayName: 'az devops login'
-- bash: |
-        echo "Bumping windows VHD base image version"
-        /bin/bash vhdbuilder/scripts/windows/automate_version_bump.sh $MAPPED_GITHUB_PAT '${{ parameters.GithubUserName }}' "$CHERRY_PICK_COMMIT_ID"
-  env:
-        MAPPED_GITHUB_PAT: $(GITHUB_PAT)
-  displayName: 'Image Version Bumping'
-  condition: eq('${{ parameters.ImageBump }}', true)
-  
\ No newline at end of file
+  - stage: ev2_artifact
+    dependsOn: []
+    condition: eq('${{ parameters.ArtifactTrigger }}', true)
+    jobs:
+    - job: ev2
+      timeoutInMinutes: 180
+      steps:
+        - bash: |
+                az extension add -n azure-devops
+                echo $MAPPED_ADO_PAT | az devops login --organization=https://dev.azure.com/msazure
+                az devops configure --defaults organization=https://dev.azure.com/msazure project=CloudNativeCompute
+          env:
+                MAPPED_ADO_PAT: $(ADO_PAT)
+          displayName: 'az devops login'
+        - bash: |
+                echo "Triggering windows VHD build ev2 artifacts pipeline"
+                az login --identity
+                pipeline_id="322622"
+                echo "Build ID for the release is $BUILD_ID"
+                run_id=$(az pipelines run --id $pipeline_id --variables VHD_PIPELINE_RUN_ID=$BUILD_ID | jq -r '.id')
```

Comment:
where are we specifying the source artifacts to run this off of? the naming is usually based on the name of those artifacts - wondering if something funky is going on with that specifically

Reply:
The source artifacts are set in pipeline UI variable `BUILD_ID`. the release name format is $(Build.BuildNumber)-$(rev:r). `$(Build.BuildNumber)` was the funky thing. It's empty.

Reply:
btw --artifact-metadata-list "ev2_artifacts=$RUN_ID" the `RUN_ID` is the id for **Build EV2 artifacts** not the VHD build id (`VHD_PIPELINE_RUN_ID`) if that's confusing. Techincally no "name" is passed directly, it should be deduced from id (either `BUILD_ID` or `run_id`)